### ISSUE TITLE
tripledhikm redunantly and confusingly needs 6 parameters while it works really on only 4

### ISSUE THREAD

**Issue by stef** [cfrg/draft-irtf-cfrg-opaque#321]
```
TripleDHIKM(sk1, pk1, sk2, pk2, sk3, pk3)
```
is later called with:
```
5. ikm = TripleDHIKM(server_private_keyshare, ke1.client_keyshare,
                    server_private_key, ke1.client_keyshare,
                    server_private_keyshare, client_public_key)
```
which is just a combination of eskS, skS and pkU, and epkU, i think the current way is very confusing and a source of silent coding errors.

**Comment by stef** [comment]
i realized this is a way to have only one function doing the 3dh instead of one for the server and one for the client. although i do understand this ,i still believe that the caller should not be exposed to the semantics of the underlying function.  it definitely does not help readability of the code the cognitive costs of reading and getting the params right is in my opinion not worth having only one function.

**Comment by bytemare** [comment]
I did that in my first implementation and even proposed that in a PR here: have one generic function and switch the computation based on a role argument (client or server). I don't remember why we didn't keep it.

I ended up not doing it anymore because it added a slight complexity to the code, and the KE would fail anyway if not implemented correctly. For the spec, only implementors will need to read this thoroughly.

I'm not saying I don't agree with you :) I understand your argument!
